staging/src/k8s.io/component-helpers/scheduling/corev1/resource_names_test.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package corev1
+
+import (
+	"fmt"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestIsPrefixedNativeResourceName(t *testing.T) {
+	testCases := []struct {
+		resourceName v1.ResourceName
+		expectVal    bool
+	}{
+		// explicit native resource names (kubernetes.io is explicitly mentioned)
+		{
+			resourceName: "pod.alpha.kubernetes.io/opaque-int-resource-foo",
+			expectVal:    true,
+		},
+		{
+			resourceName: "kubernetes.io/resource-foo",
+			expectVal:    true,
+		},
+		// implicit native resource names are rejected
+		{
+			resourceName: "foo",
+			expectVal:    false,
+		},
+		{
+			resourceName: "a/b",
+			expectVal:    false,
+		},
+		{
+			resourceName: "",
+			expectVal:    false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("resourceName input=%s, expected value=%v", tc.resourceName, tc.expectVal), func(t *testing.T) {
```

Comment:
I would just put tc.resourceName as test name.
A simpler test name is useful for filtering tests.

Same for the other tests

Reply:
Some `tc.resourceName` are empty strings. In some cases the field is not present. I have added new `name` field with describes what the test is intended to exercise.